### PR TITLE
SDK v2 changes - 2

### PR DESCRIPTION
**Pull request by HunaidHanfee-MSFT**

### CODE REVIEW

File: samples/tab-sso/nodejs/src/client/views/auth-start.pug [hunk 1]
```diff
@@ -5,33 +5,35 @@ html
...
+                    scope: "https://graph.microsoft.com/User.Read email openid profile offline_access",
+                    redirect_uri: window.location.origin + "/auth-end",
+                    nonce: _guid(),
+                    state: state,
+                    login_hint: context.user.loginHint,
+                };
+                // Go to the AzureAD authorization endpoint (tenant-specific endpoint, not "common")
+                // For guest users, we want an access token for the tenant we are currently in, not the home tenant of the guest. 
+                let authorizeEndpoint = `https://login.microsoftonline.com/${context.user.tenant.id}/oauth2/v2.0/authorize?${toQueryString(queryParams)}`;
+                window.location.assign(authorizeEndpoint);
```
Comment (v-chetsh) [inline samples/tab-sso/nodejs/src/client/views/auth-start.pug:33]: Kindly add proper spacing and line breaks.

**Review by v-chetsh — CHANGES_REQUESTED** [review]
Some cosmetic changes needs to be addressed.

File: samples/msgext-search-sso-config/appPackage/manifest.json [hunk 2]
```diff
@@ -111,6 +111,6 @@
   ],
   "webApplicationInfo": {
     "id": "{TODO: MicrosoftAppId}",
-    "resource": "api://{TODO: SiteUrlDomain}/botid-{TODO: MicrosoftAppId}/{TODO: MicrosoftAppId}"
+    "resource": "api://{TODO: SiteUrlDomain}/botid-{TODO: MicrosoftAppId}"
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/msgext-search-sso-config/appPackage/manifest.json:114]: It's the same overall the file

File: samples/tab-sso/csharp/TeamsTabSSO/Views/Auth/Start.cshtml [hunk 3]
```diff
@@ -7,32 +7,33 @@
...
+                        response_mode: "fragment",
+                        resource: "https://graph.microsoft.com/",
+                        redirect_uri: window.location.origin + "/Auth/End",
+                        nonce: _guid(),
+                        state: state,
+                        // login_hint pre-fills the username/email address field of the sign in page for the user,
+                        // if you know their username ahead of time.
+                        login_hint: context.user.userPrincipalName
+                    };
+                    // Go to the AzureAD authorization endpoint
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/tab-sso/csharp/TeamsTabSSO/Views/Auth/Start.cshtml:31]: Done

File: samples/tab-sso/nodejs/src/client/views/auth-start.pug (hunk 1 above)
Comment (HunaidHanfee-MSFT) [reply samples/tab-sso/nodejs/src/client/views/auth-start.pug:35]: Done

File: samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Home/SilentConfigureTab.cshtml [hunk 4]
```diff
@@ -42,11 +40,24 @@
...
+            microsoftTeams.pages.config.registerOnSaveHandler(function (saveEvent) {
+                var radios = document.getElementsByName('notificationType');
+
+                microsoftTeams.pages.config.setConfig({
+                    entityId: 'Silent Authentication',
+                    contentUrl: window.location.origin + "/Home/Index",
+                    configName: 'SilentAuthentication'
+                });
+                saveEvent.notifySuccess();
+            });
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Home/SilentConfigureTab.cshtml:53]: Don

File: samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Authentication/AuthEnd.cshtml [hunk 5]
```diff
@@ -28,35 +28,35 @@
...
-                console.log(msalInstance.getTokenCache());
-                microsoftTeams.authentication.notifyFailure("Unable to login");
-                return;
-            } else
-            {
-                microsoftTeams.authentication.notifySuccess();
-            }
+            msalInstance.handleRedirectPromise(window.location.hash).then((tokenResponse) => {
+                const currentAccounts = msalInstance.getAllAccounts();
+                if (!currentAccounts || currentAccounts.length === 0) {
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/tab-channel-group-config-page-auth/csharp/ConfigTabAuthentication/Views/Authentication/AuthEnd.cshtml:51]: Added

File: samples/meetings-live-code-interview/csharp/MeetingLiveCoding/ClientApp/src/components/question.tsx [hunk 6]
```diff
@@ -15,19 +15,19 @@ const Question =(props :any)=>
...
-        
+        microsoftTeams.app.initialize();
     }, [])
 
     React.useEffect(() => {
-        microsoftTeams.initialize();
-        microsoftTeams.getContext((context: any) => {
-            setMeetingId(context.meetingId)
-            getLatestEditorValue(questionNumber, context.meetingId).then((res: any) => {
-                if(res.data.value != null && res.data.value != "")
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/meetings-live-code-interview/csharp/MeetingLiveCoding/ClientApp/src/components/question.tsx:27]: Renamed to result

File: samples/meetings-live-code-interview/nodejs/ClientApp/src/components/question.tsx [hunk 7]
```diff
@@ -15,13 +15,14 @@ const Question = (props: any) => {
...
-        microsoftTeams.getContext((context:any) =>{
-            setMeetingId(context.meetingId)
-            getLatestEditorValue(questionNumber,context.meetingId).then((res:any) =>{
-                setData(res.data.value);
-            })
-        })
+        microsoftTeams.app.initialize().then(() => {
+            microsoftTeams.app.getContext().then((context:any) =>{
+                setMeetingId(context.meeting.id)
+                getLatestEditorValue(questionNumber,context.meetingId).then((res:any) =>{
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/meetings-live-code-interview/nodejs/ClientApp/src/components/question.tsx]: Renamed to result

File: samples/graph-chat-lifecycle/nodejs/src/views/configure.html [hunk 8]
```diff
@@ -7,19 +7,19 @@
...
+    microsoftTeams.app.notifySuccess();
+    microsoftTeams.pages.config.registerOnSaveHandler(function (saveEvent) {
+      microsoftTeams.pages.config.setConfig({
+        entityID: "Chat Lifecycle App",
+        contentUrl: `${window.location.origin}/chatLifecycle`,
+        suggestedTabName: "Chat Lifecycle App",
+        websiteUrl: `${window.location.origin}/chatLifecycle`,
+      });
+      saveEvent.notifySuccess();
     });
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/graph-chat-lifecycle/nodejs/src/views/configure.html:27]: Added

File: samples/graph-chat-lifecycle/nodejs/src/views/auth-start.html [hunk 9]
```diff
@@ -6,33 +6,35 @@
...
+                    scope: "https://graph.microsoft.com/User.Read email openid profile offline_access",
+                    redirect_uri: window.location.origin + "/auth/auth-end",
+                    nonce: _guid(),
+                    state: state,
+                    login_hint: context.user.loginHint,
+                };
+                // Go to the AzureAD authorization endpoint (tenant-specific endpoint, not "common")
+                // For guest users, we want an access token for the tenant we are currently in, not the home tenant of the guest. 
+                let authorizeEndpoint = `https://login.microsoftonline.com/${context.user.tenant.id}/oauth2/v2.0/authorize?${toQueryString(queryParams)}`;
+                window.location.assign(authorizeEndpoint);
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/graph-chat-lifecycle/nodejs/src/views/auth-start.html:36]: Added

File: samples/app-sso/csharp/App SSO Sample/wwwroot/js/auth.js [hunk 10]
```diff
@@ -27,50 +27,44 @@ function requestConsent() {
...
-            getUserInfo(context.userPrincipalName);
-            getUserPhoto(context.userPrincipalName);
+            getUserInfo(context.user.userPrincipalName);
+            getUserPhoto(context.user.userPrincipalName);
         });
     });
 }
 
 function getToken() {
+    alert("CALLLING");
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/app-sso/csharp/App SSO Sample/wwwroot/js/auth.js]: Removed

File: samples/graph-chat-lifecycle/csharp/ChatLifecycle/Views/Auth/Start.cshtml [hunk 11]
```diff
@@ -7,32 +7,33 @@
...
+                        response_type: "id_token token",
+                        response_mode: "fragment",
+                        resource: "https://graph.microsoft.com/",
+                        redirect_uri: window.location.origin + "/Auth/End",
+                        nonce: _guid(),
+                        state: state,
+                        // login_hint pre-fills the username/email address field of the sign in page for the user,
+                        // if you know their username ahead of time.
+                        login_hint: context.user.loginHint
+                    };
```
[the remark this answers is not in the document]
Comment (HunaidHanfee-MSFT) [reply samples/graph-chat-lifecycle/csharp/ChatLifecycle/Views/Auth/Start.cshtml:30]: Done

**Review by v-chetsh — APPROVED** [review]
Changes looks fine, approving it.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**